Source/ProxyConnect.swift

```diff
@@ -0,0 +1,428 @@
+        self.url = url;
+        if ["wss", "https"].contains(url.scheme) {
+            self.secure = true
+        }
+    }
+    
+    public func openNetworkStream(completionHandler:(error:NSError?, readStream:NSInputStream?, writeStream : NSOutputStream?) -> Void) {
+        connectDoneHandler = completionHandler
+        configureProxy()
+    }
+    
+    private func didConnect() {
+        ProxyFastLog("_didConnect, return streams");
+        if secure {
+            if httpProxyHost != nil {
+                // Must set the real peer name before turning on SSL
+                ProxyFastLog("proxy set peer name to real host \(url.host)")
+                outputStream?.setProperty(url.host,forKey:"_kCFStreamPropertySocketPeerName")
+            }
+        }
+        
+        receivedHTTPHeaders = nil
+        
+        if let stream = inputStream {
+            CFReadStreamSetDispatchQueue(stream, nil)
+        }
+        inputStream?.delegate = nil
+        outputStream?.delegate = nil
+        if connectDoneHandler != nil {
+            connectDoneHandler!(error:nil, readStream:inputStream, writeStream:outputStream)
+        }
+        
+    }
+    
+    private func connectionFailed(error: NSError?) {
+        ProxyFastLog("_failWithError, return error");
+        var err = error
+        if err == nil  {
+            err  = NSError(domain: "Proxy", code : Int(500),
+                           userInfo : [NSLocalizedDescriptionKey:"Proxy Error"])
+        }
+        
+        receivedHTTPHeaders = nil
+        
+        outputStream?.delegate = nil
+        inputStream?.delegate = nil
+        if let stream = inputStream {
+            CFReadStreamSetDispatchQueue(stream, nil)
+            stream.close()
+        }
+        if let stream = outputStream {
+            CFWriteStreamSetDispatchQueue(stream, nil)
+            stream.close()
+        }
+        outputStream = nil
+        inputStream = nil
+        if connectDoneHandler != nil {
+            connectDoneHandler!(error:err, readStream:nil, writeStream:nil)
+        }
+    }
+    
+    // get proxy setting from device setting
+    private func configureProxy () {
+        ProxyFastLog("configureProxy");
+        var hURL:NSURL? = url;
+        if let host = url.host {
+            if secure {
+                hURL = NSURL(string : "https://"+host)
+            } else {
+                hURL = NSURL(string : "http://"+host)
+            }
+        }
+        if hURL == nil {
+            hURL = url
+        }
+        if let proxySettings: NSDictionary = CFNetworkCopySystemProxySettings()?.takeRetainedValue() {
+            let proxies : NSArray = CFNetworkCopyProxiesForURL(hURL!, proxySettings).takeRetainedValue()
+            if proxies.count == 0 {
+                ProxyFastLog("configureProxy no proxy")
+                openConnection()
+                return
+            }
+            
+            let settings:NSDictionary = proxies[0] as! NSDictionary
```

Comment:
Oh, I see.

I did some changes please take a look.

I just came from the objC side, still not very familiar of these finer 
points of Swift programming :(

Thanks,

Dong

On 6/4/16 12:12 PM, Erik Little wrote:
>
> In Source/ProxyConnect.swift 
> <https://github.com/daltoniam/Starscream/pull/202#discussion_r65804317>:
>
> > +            } else {
> > +                hURL = NSURL(string : "http://"+host)
> > +            }
> > +        }
> > +        if hURL == nil {
> > +            hURL = url
> > +        }
> > +        if let proxySettings: NSDictionary = CFNetworkCopySystemProxySettings()?.takeRetainedValue() {
> > +            let proxies : NSArray = CFNetworkCopyProxiesForURL(hURL!, proxySettings).takeRetainedValue()
> > +            if proxies.count == 0 {
> > +                ProxyFastLog("configureProxy no proxy")
> > +                openConnection()
> > +                return
> > +            }
> > +
> > +            let settings:NSDictionary = proxies[0] as! NSDictionary
>
> Like here, you don't need to explicitly type since you're doing |as!|. 
> You're also inconsistent with your |:|s. The style used should be |let 
> x: Type|
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/daltoniam/Starscream/pull/202/files/ccffac6966ff55fac019123c8198c3baeabdc094#r65804317>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AHRzfOEK474MrgV654uDYX6L5EOT0_cwks5qIaP9gaJpZM4IuEmH>.
>